src/character_attire.cpp

```diff
@@ -1945,6 +1950,171 @@ void outfit::absorb_damage( Character &guy, damage_unit &elem, bodypart_id bp,
+            // The item has intercepted the splash to protect its wearer,
+            // now we roll to see if it's affected.
+            guy.add_msg_if_player( m_warning, _( "Your %1s is splashed with %2s %3s." ),
+                                   armor.tname(), get_liquid_descriptor( liquid_remaining ), liquid_name );
+            // A droplet of acid or bile are less likely to ruin a shirt than a whole bucket.
+            if( rng( 1, 200 - armor.breathability( bp ) ) < liquid_remaining ) {
+                // Apply filth to the item. Currently hardcoded because we don't have other item
+                // flags that would make sense for this. It gets its own probability roll here
+                // because it can't use armor_absorb's.
+                if( sp.has_flag( spell_flag::MAKE_FILTHY ) && !armor.has_flag( flag_INTEGRATED ) &&
+                    !armor.has_flag( flag_SEMITANGIBLE ) && !armor.has_flag( flag_PERSONAL ) &&
+                    !armor.has_flag( flag_AURA ) &&
+                    rng( 1, 200 - armor.breathability( bp ) ) < liquid_remaining ) {
+                    add_msg_if_player_sees( guy, m_warning, _( "%1s %2s is covered in filth!" ), guy.disp_name( true,
+                                            true ),
+                                            armor.tname() );
+                    armor.set_flag( json_flag_FILTHY );
+                    guy.on_worn_item_soiled( armor );
+                }
+                // If this is an armor-damaging liquid, the damage is relative to fluid_remaining
+                // and the coverage of the item.
+                if( damage.amount >= 1.0f && damage_armor ) {
+                    const std::string pre_damage_name = armor.tname();
+                    bool destroy = false;
+                    item_armor_enchantment_adjust( guy, damage, armor );
+                    if( ignite && damage.amount >= 1.0f ) {
+                        int fire_intensity = std::ceil( intensity * ( liquid_remaining / liquid_amount ) );
+                        if( fire_intensity >= 1 ) {
+                            fire_data frd{ fire_intensity };
+                            destroy = !armor.has_flag( flag_INTEGRATED ) && armor.burn( frd );
+                            int fuel = roll_remainder( frd.fuel_produced );
+                            if( fuel > 0 ) {
+                                guy.add_effect( effect_onfire, time_duration::from_turns( fuel + 1 ), bp, false, 0, false,
+                                                true );
+                            }
+                        }
+                    }
+                    if( !destroy ) {
+                        // The roll here is 0 because we already rolled to see if the attack hurts the armor.
+                        // Not -1 so that we can't splash items with 0 coverage.
+                        if( secondary_sbp != sub_bodypart_id() ) {
```

Comment:
Neither `secondary_sbp` nor `sbp` is guaranteed to be non-null at this point, this whole if block will likely result in a crash or undefined behavior.